File: src/test/java/io/zbus/examples/mq/ConsumerExample.java
package io.zbus.examples.mq;

import io.zbus.mq.Broker;
import io.zbus.mq.Consumer;
import io.zbus.mq.Message;
import io.zbus.mq.ZbusBroker;

public class ConsumerExample { 
	@SuppressWarnings("resource")
	public static void main(String[] args) throws Exception {  
		
		Broker broker = new ZbusBroker("127.0.0.1:15555");    
		
		Consumer consumer = new Consumer(broker, "MyTopic");  
		consumer.declareTopic();
		
		while(true){
			Message message = consumer.take();
			System.out.println(message.getOffset());
		}  
	}
}

File: clients/DotNET/Zbus.Examples/Mq/ConsumerExample.cs
using log4net;
using System;
using System.Threading.Tasks;
using Zbus.Mq;

namespace Zbus.Examples
{
    class ConsumerExample
    {
        private static readonly ILog log = LogManager.GetLogger(typeof(ConsumerExample));
        static void Main(string[] args)
        {
            log.Info("begin test");

            Broker broker = new Broker();
            broker.AddTracker("localhost:15555");

            Consumer c = new Consumer(broker, "MyTopic");
            c.MessageReceived += (msg, client) => {
                Console.WriteLine(msg);
            };
            c.Start();

            Console.WriteLine("done");
            Console.ReadKey();
        }
    }
}
